### PR TITLE
implementation of fortify on demand scan

### PR DESCRIPTION
**Pull request by kaanaktas**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk.gov.hmcts.fodscanner/FailureLevel.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.fodscanner;
+
+public enum FailureLevel {
```
Comment (MSancaktutar) [inline src/main/java/uk.gov.hmcts.fodscanner/FailureLevel.java:3]: rename to pls Severity

File: src/main/java/uk.gov.hmcts.fodscanner/PropertiesSingleton.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    public static PropertiesSingleton getInstance() {
+        return Initializer.INSTANCE;
+    }
+
+    public String getValue(String key) {
+        return Validate.notNull(properties.getProperty(key), "Property key `%s` is required", key);
+    }
+
+    private Properties getPropertiesFromFile() throws Exception {
```
Comment (MSancaktutar) [inline src/main/java/uk.gov.hmcts.fodscanner/PropertiesSingleton.java:33]: rename to getPropertiesFromResource please

File: src/main/java/uk.gov.hmcts.fodscanner/FailureLevel.java (hunk 1 above)
Comment (kaanaktas) [reply src/main/java/uk.gov.hmcts.fodscanner/FailureLevel.java]: done

File: src/main/java/uk.gov.hmcts.fodscanner/PropertiesSingleton.java (hunk 2 above)
Comment (kaanaktas) [reply src/main/java/uk.gov.hmcts.fodscanner/PropertiesSingleton.java]: done

File: src/test/java/uk/gov/hmcts/fortifyclient/FortifyClientTest.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    @Test
+    void should_fail_scan_with_wrong_credentials() throws Exception {
+        FortifyClientConfig config = FortifyClientConfig.getNewDefaultInstance();
+        FortifyClient fortifyClient = new FortifyClient(config);
+        Exception e = Assertions.assertThrows(FortifyClientException.class,
+                fortifyClient::requestScanAndGetResults);
+        Assertions.assertTrue(e.getMessage().contains("code=401, message=Unauthorized"));
+    }
+
+}
```
Comment (MSancaktutar) [inline src/test/java/uk/gov/hmcts/fortifyclient/FortifyClientTest.java:17]: eol at eof, always

File: src/test/java/uk/gov/hmcts/fortifyclient/ScanReportTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        counts.put(Severity.MEDIUM, 0);
+        counts.put(Severity.LOW, 0);
+        assertTrue(scanReport.hasAnyIssuesAtOrAbove(Severity.CRITICAL));
+        assertTrue(scanReport.hasAnyIssuesAtOrAbove(Severity.HIGH));
+        assertTrue(scanReport.hasAnyIssuesAtOrAbove(Severity.MEDIUM));
+        assertTrue(scanReport.hasAnyIssuesAtOrAbove(Severity.LOW));
+    }
+
+    @Test
+    void isSuccessful() {
```
Comment (MSancaktutar) [inline src/test/java/uk/gov/hmcts/fortifyclient/ScanReportTest.java:47]: better name for test methods that tells what they are testing. pls see examples from ccd repos for good naming examples.

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
Please replace all std outs with debug / info / error logs, whichever more appropriate at each point.

**Review by MSancaktutar — COMMENTED** [review]
A lot more unit and integration tests would be required, but that can be done under a tech debt ticket, to be actioned by any contributor later on. I'll be happy to release this if comments before all fully applied.

File: src/test/java/uk/gov/hmcts/fortifyclient/FortifyClientTest.java (hunk 3 above)
Comment (kaanaktas) [reply src/test/java/uk/gov/hmcts/fortifyclient/FortifyClientTest.java:17]: done

File: src/test/java/uk/gov/hmcts/fortifyclient/ScanReportTest.java (hunk 4 above)
Comment (kaanaktas) [reply src/test/java/uk/gov/hmcts/fortifyclient/ScanReportTest.java:47]: done

File: src/test/java/uk/gov/hmcts/fortifyclient/ScanReportTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        Map<Severity, Integer> counts = new HashMap<>();
+        ScanReport scanReport = new ScanReport(clientConfig, counts);
+        counts.put(Severity.CRITICAL, 3);
+        counts.put(Severity.HIGH, 5);
+        counts.put(Severity.MEDIUM, 1);
+        counts.put(Severity.LOW, 6);
+        scanReport.printToDefaultHtml();
+
+    }
+}
```
Comment (MSancaktutar) [inline src/test/java/uk/gov/hmcts/fortifyclient/ScanReportTest.java:84]: eol at eof

File: src/test/java/uk/gov/hmcts/fortifyclient/FolderZipperTest.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    @Test
+    void should_successfully_zip() throws Exception {
+
+        FortifyClientConfig config = FortifyClientConfig.getNewDefaultInstance();
+
+        FolderZipper zipper = new FolderZipper();
+        zipper.zip(new File("."), config.getExcludePatterns());
+    }
+
+}
```
Comment (MSancaktutar) [inline src/test/java/uk/gov/hmcts/fortifyclient/FolderZipperTest.java:18]: eol at eof

File: src/main/resources/ReportTemplate.html [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+			<td>---</td>
+			<td>---</td>
+		</tr>
+		<tr>
+			<td>Scan Result Acceptable</td>
+			<td>[AcceptableStatus]</td>
+		</tr>
+	</table>
+</body>
+</html>
```
Comment (MSancaktutar) [inline src/main/resources/ReportTemplate.html:50]: eol at eof

**Review by MSancaktutar — APPROVED** [review]
LGTM

**Comment by MSancaktutar** [comment]
Need more unit and integration tests and a few more minor improvements could be made, but they can be done by a next contributor. Good enough to release as an initial, stable release.

**Pull request merged**